Wikipedia/Code/PageHistoryViewController.swift

```diff
@@ -296,8 +299,23 @@ class PageHistoryViewController: ColumnarCollectionViewController {
                 self.appendSections(from: results)
                 self.pageHistoryFetcherParams = results.getPageHistoryRequestParameters(self.pageURL)
                 self.batchComplete = results.batchComplete()
-                self.isLoadingData = false
-                self.collectionView.reloadData()
+                let completeLoad = {
+                    self.isLoadingData = false
+                    self.collectionView.reloadData()
+                }
+
+                if let revisionToScrollTo = self.revisionToScrollTo {
+                    let hasScrollRevisionBeenLoaded = (revisionToScrollTo >= (results.lastRevision?.revisionID ?? 0))
+                    if !hasScrollRevisionBeenLoaded {
+                        // Load through the target revision
+                        self.getPageHistory()
```

Comment:
@mcleinman This is fine though I think maybe we should have a max number of times this loops through just as a safety net in case it gets in some wacky state and it's calling endlessly. Maybe 10 pages or so?

Reply:
I added the `let hasScrollRevisionBeenLoaded = (revisionToScrollTo >= (results.lastRevision?.revisionID ?? 0)` line to hopefully have a couple safeties: If `lastRevision` doesn't come in, we stop (because we use `0`. And using `>=` even though a simple `>` should work - that way if we somehow don't load the actual revision but load past the time it should be in, we stop.

If we want to load past the first page, I'm a tad reluctant to cut it off arbitrarily, especially at something like 10 pages. Would 50 or 100 be okay? That's something that is well beyond where any user would likely scroll to, but also isn't an incredible amount of network requests to make in that disaster "something went horribly wrong" scenario.

Reply:
Yeah, to be clear I was thinking of keeping your existing code intact, and just additionally keeping track of the number of recursive callbacks and don't loop again if it reaches a certain number (I'm fine with 50 or 100), so like `if !hasScrollRevisionBeenLoaded && numIterations < 50) { self.getPageHistory() }...`. But I agree your code does look fail-safe - the only crazy scenario I could think of it messing up is if revisionIDs don't decrement as assumed the further back you go, but I have no reason to think that could happen. Anyways up to you, you can add the increment & check if you'd like but I'm fine with leaving it as-is too.